### DISCUSSION TITLE
Dynamic Thresholds

### DISCUSSION THREAD

**Discussion by elasticdotventures** [grafana/grafana#70518] [Dashboards, Panels & Visualizations]
According to visualizations documentation:

https://grafana.com/docs/grafana/latest/panels-visualizations/configure-thresholds/

Thresholds currently support absolute & percentage.
I'm looking for a "Dynamic" threshold.

The specific use case is documented in slack here:
https://grafana.slack.com/archives/C043CEX9MBK/p1687440476894409

I need/want to be able to load a value that isn't stored in the json, 
the most suitable candidate in my mind for how to implement this would be to load the most current value from a redis key, or some other form of high speed memory cache.  

It would also be acceptable to update these values using the API - in that example the value for a dynamic threshold might be derived from a recording rule, which is a better approach since it's a feature that already exists. 

This would allow the threshold for the graph to be dynamically updated without republishing.
I'm not sure if this has ever been discussed, or if a PR for such a feature would be accepted.

I'd also like to be able to use these dynamic values for alert manager, but I suspect this is the wrong forum to say that, I just want to make it clear that the dynamic threshold is intended for a dashboard that is tied to an alarm & associated runbook for an alert.  
I want to show all the data on the graph, and when the threshold value is exceeded for a period, then fire the alarm in alert manager.

**Comment by leeoniya** [comment]
you can use "Config from query results", but it's rather limited to using a single threshold, and the colors are hard-coded to green/red. (see https://github.com/grafana/grafana/issues/56425)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). This looks to be answered, so I will close it. Feel free to make an [issue](https://github.com/grafana/grafana/issues) if find additional bugs or have a feature request.
